packages/cicd/github-actions/src/section/github-actions-section-generator.adapter.ts

```diff
@@ -1,20 +1,34 @@
-import { Repository } from '@ci-dokumentor/core';
+import { RepositoryProvider } from '@ci-dokumentor/core';
 import {
   FormatterAdapter,
   SectionGeneratorAdapter,
   SectionIdentifier,
+  SectionOptions,
 } from '@ci-dokumentor/core';
 import { GitHubAction, GitHubActionsManifest, GitHubWorkflow } from 'src/github-actions-parser.js';
 
 export abstract class GitHubActionsSectionGeneratorAdapter
   implements SectionGeneratorAdapter<GitHubActionsManifest> {
   abstract getSectionIdentifier(): SectionIdentifier;
 
+  /**
+   * Generate section content using repository provider for on-demand data fetching
+   */
   abstract generateSection(
     formatterAdapter: FormatterAdapter,
     manifest: GitHubActionsManifest,
-    repository: Repository
-  ): Buffer;
+    repositoryProvider: RepositoryProvider
+  ): Promise<Buffer>;
+
+  /**
+   * Optional: provide CLI option descriptors specific to this section generator
+   */
+  getSectionOptions?(): Record<string, any>;
```

Comment:
Do not declare optional method, methods are always mandatory be can return an empty record